### ISSUE TITLE
Fix up Cached Buffer in Fruit Salad branch/dir.

### ISSUE THREAD

**Issue by LiamKarlMitchell** [LiamKarlMitchell/InfiniteSky#189]
It should not have this. variables all over the place those things are not accessed outside it so should just be var.

Only a few of the things need to persist / be on this.

**Comment by LiamKarlMitchell** [comment]
Also it might be better to have a fixed sized buffer rather than concat.
Since that would need to allocate memory each time right?

**Comment by Ane2** [comment]
Yeah some limit to the buffer would be nice. Concat still would be used since we are processing data of that.
